### ISSUE TITLE
`auto-save` clobbers marks

### ISSUE THREAD

**Issue by gegnew** [okuuva/auto-save.nvim#44]
yup, just that. `auto-save` removes all marks every time it runs.

Any way to avoid this?

Cheers!

**Comment by primeapple** [comment]
I just tried, for me marks stay. Please post your config and a reliable Steps To Reproduce.

It might be that you have other autocommands that depend on save events that are triggered by the auto save. In this case you might try `noautocmd = true` in your config.

**Comment by primeapple** [comment]
Any answer on this @gegnew ? Otherwise I'd like to close it.

**Comment by simonmandlik** [comment]
Hi @primeapple, I noticed that only `[` and `]` marks are removed (as was the case in the [upstream](https://github.com/pocco81/auto-save.nvim/issues/79))

Try this MWE:

```lua
local root = vim.fn.fnamemodify("./.repro", ":p")
for _, name in ipairs({ "config", "data", "state", "cache" }) do
    vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
    vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

local plugins = {
    -- "okuuva/auto-save.nvim",
    -- opts = {},
}

require("lazy").setup(plugins, { root = root .. "/plugins", })
```

Try (un)commenting the two lines in `plugins`, deleting something in the middle of a buffer and then pressing `']` and `'[`. It should take me to the changed lines, but when autosave is active, it takes me to the beginning and end of the buffer.

**Comment by primeapple** [comment]
Yeah, I can reproduce that.

To me it seems that the `]` and `[` marks get reset when saving the buffer. This also happens when manually saving via `:w` so that seems to be the bug (feature ;-)).

So if you find a way to get it working that manual saving doesn't clear the `[` and `]` marks I'm happy to implement it here. I tried `noautocmd` but no luck :(

**Comment by simonmandlik** [comment]
Perhaps use `:lockmarks w` instead of just `:w`? [source](https://stackoverflow.com/questions/63912328/keep-and-marks-when-i-save-vim)

**Comment by primeapple** [comment]
Opened a PR for this. I tested it locally and it seems to work. But please reassure yourself.

**Comment by primeapple** [comment]
Please confirm, that the commit changed it for you, then feel free to close @simonmandlik

**Comment by okuuva** [comment]
Fixed in #53.
